WordPress/gutenberg
# Block Support: Add text decoration block support using CSS variables

## Description

This adds block support for text decoration styles and is a continuation from discussion around https://github.com/WordPress/gutenberg/pull/25641 around adding font style block support.

#### Changes Included
* Adds block support for text decoration
* Allows for preset text decoration options and uses CSS variables
* Adds new icon for "underline"
* Updates navigation block to opt-in to text decoration support for easier testing

#### Notes
* The intention is when this and text transform block support (#26060 ) are both merged a quick update will be made to include both sets of controls so they display on a single row within the typography controls.

## How has this been tested?
Manually tested using heading block.

#### Test Instructions

1. Add a navigation block to a post and select it.
2. You should see new text decoration buttons under the Inspector Controls > Typography section.
3. Toggle the various decoration buttons and confirm the block updates visually as you'd expect.
4. With a text decoration toggled on, inspect the navigation block in dev tools and confirm that the block includes an inline style using `var()` and an appropriate CSS variable.
5. Save the post and view on the frontend.
6. The same text decoration choice should be reflected on the frontend block's inline styles.

## Screenshots 
![TextDecorationBlockSupport](https://user-images.githubusercontent.com/60436221/96824536-6350d080-1472-11eb-87af-e779dca02a7b.gif)

## Types of changes
Enhancement

## Next Steps
- [ ] Update tests in `class-block-supported-styles-test.php` if needed after approach confirmed.

## Checklist:
